client/state/reader/conversations/reducer.js

```diff
@@ -42,8 +43,32 @@ export const items = createReducer(
 			return newState;
 		},
 		[ READER_CONVERSATION_UPDATE_FOLLOW_STATUS ]: ( state, action ) => {
+			const stateKey = key( action.payload.siteId, action.payload.postId );
+
+			// If followStatus is null, remove the key from the state map entirely
+			if ( action.payload.followStatus === null ) {
```

Comment:
we should check against `READER_CONVERSATION_NOT_FOLLOWING` instead of against null.